pkg/render/namespaces.go

```diff
@@ -17,51 +17,8 @@ package render
 import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-
-	operator "github.com/tigera/operator/pkg/apis/operator/v1"
-	"github.com/tigera/operator/pkg/common"
-)
-
-const (
-	TigeraPrometheusNamespace = "tigera-prometheus"
 )
 
-func Namespaces(cr *operator.Installation, openshift bool, pullSecrets []*corev1.Secret) Component {
```

Comment:
@tmjd  I opened this PR to show / ask why node is creating namespace resources using this Component interface instead of just running these four lines of code like the other controllers:

```
		objs = append(objs, createNamespace(TigeraPrometheusNamespace, c.openshift))
	if len(c.pullSecrets) > 0 {
		objs = append(objs, copyImagePullSecrets(c.pullSecrets, TigeraPrometheusNamespace)...)
	}
```

Do the other controllers need to be updated, is the Installation controller outdated, or is there some other special case that the Installation resource needs to account for?

Reply:
The core was written a little differently than the other controllers is probably the main reason.